src/dpp/guild.cpp

```diff
@@ -112,6 +112,27 @@ std::string guild_member::get_mention() const {
 
 guild_member& guild_member::set_nickname(const std::string& nick) {
 	this->nickname = nick;
+	this->flags |= gm_nickname_action;
+	return *this;
+}
+
+guild_member& guild_member::add_role(dpp::snowflake role_id) {
+	roles.emplace_back(role_id);
+	flags |= gm_roles_action;
+	return *this;
+}
+
+guild_member& guild_member::remove_role(dpp::snowflake role_id) {
+	roles.erase(std::remove_if(roles.begin(), roles.end(), [&role_id](dpp::snowflake role) {
```

Comment:
Can just use `std::remove` here instead of using `std::remove_if` with a lambda

Reply:
did you mean std::erase, seems to be C++20 only?

https://en.cppreference.com/w/cpp/container/vector/erase2